### PR TITLE
updated functions to ES6 plus var->let

### LINKED ISSUE #612: Update L.FreeRotateHandle method creation to ES6 syntax
## First Time?

This is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, consider leaving this one for someone new, and looking through our general [help wanted](https://github.com/publiclab/Leaflet.DistortableImage/issues) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

## The problem

We just added support for ES6 in our library, and would like to start transitioning our code to use this syntax.

Let's update the method creation syntax in the `src/edit/handles/FreeRotateHandle.js` file from ES5 to ES6.

## Solution

**Where to find the relevant lines of code**: 

For the 2 functions in `src/edit/handles/FreeRotateHandle.js`, remove the `function` keyword and attach the method arguments to the method name instead.

Here is an example of how this will look for a different method. These ones should follow suit:

ES5
```JS
radiansToDegrees: function(angle) {
  return (angle * 180) / Math.PI;
},
```

ES6
```JS
radiansToDegrees(angle) {
  return (angle * 180) / Math.PI;
},
```

Thanks!!

## Step by Step
- [ ] Claim this issue with a comment here, below, and ask any clarifying questions you need
- [ ] Fork the repository and set it up locally following the main repo README instructions [https://github.com/publiclab/Leaflet.DistortableImage](https://github.com/publiclab/Leaflet.DistortableImage)
 - [ ] Create a new feature branch with a unique name descriptive to the issue
- [ ] Try to fix the issue following the steps above, but even before you're done, you can:
  commit your changes to your branch and start a pull request (see [contributing to Public Lab software](https://publiclab.org/wiki/contributing-to-public-lab-software)) but mark it as "in progress" if you have questions or if you haven't finished
- [ ] Reference this issue in your pull request body
- [ ] Once you submit your pull request, if there's an additional checklist provided for getting it merged, get those boxes checked off. Either way, mention me @sashadev-sky for a review.

### Please keep us updated
💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

##  💬   Get help
If you need any help - here are some options: 

- Comment below
- Join our gitter chat at https://gitter.im/publiclab/publiclab

### PR DESCRIPTION
**Pull request by Manwej**
Fixes #612 (<=== Add issue number here)
 @sashadev-sky

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updates
* [x] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Manwej** [comment]
@sashadev-sky please review

File: src/edit/handles/FreeRotateHandle.js [hunk 1]
```diff
@@ -1,44 +1,48 @@
 L.FreeRotateHandle = L.EditHandle.extend({
   options: {
-    TYPE: 'freeRotate',
+    TYPE: "freeRotate",
```
Comment (sashadev-sky) [inline src/edit/handles/FreeRotateHandle.js:3]: We prefer single strings where possible

```suggestion
    TYPE: 'freeRotate',
```

File: src/edit/handles/FreeRotateHandle.js [hunk 2]
```diff
@@ -1,44 +1,48 @@
...
+        "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAklEQVR4AewaftIAAAHiSURBVMXBa3HbShgA0PMp/1sCCo8oCEpgTaCXgIXAJiDzyCJoAUTm4UVQAns1Y8+snWnTvJyeE16hkjDgDrfoNTMKcpC9UPiLSo8JyetkjEHxjPCMyoS199kFoz8Iv1HpMaN3qWDCHoegOKkkRwnJpRmroHgiPFEZ8IBekzEGxQtUEhKSS/fB7Ew4U+lxcGkVZG9QWWPSFAxBcdK59KApuA+yNwp2uEdx1GN25sZJZULSfAtm77SlbNjju6MvG75u+WHRWVR6rDVjMPsgwYyVZl3pLTpHkyYHOx8syMiayaJzlDTZ9YyaZNFVkiYH2ZUEBcVJJXVImuz6Js3Qofe59pq7DoOTILu+g+a288mCouk7/1iH4qTS+2QdDppbV1ZJmrnDXnPnc5UOs2Z0fUmTuyBr+krvSioJyUmQO0dZM7mepMkWnaNRkyrJB6uskTSjxY3Fll8bvmJwlDb83FJ8gMqAB80uyBY3Trb82PAfvjj6vuHnluIdKgMeNXOwctK5NKBoHitrb1RJeHRp5Ux4ojLg0aWMHGQvUOkxIWkKVsHsTPiNSo8HDC5lZIsgO6n0uMUdRvQuFQxB8UR4RmXC2vvsgtEfhL+o9JiQvE7GGBTPCK9QSUjoMWgKDthjDrIX+h/k0I7gth6N5gAAAABJRU5ErkJggg==",
       iconSize: [32, 32],
-      iconAnchor: [16, 16],
-    }),
+      iconAnchor: [16, 16]
```
Comment (sashadev-sky) [inline src/edit/handles/FreeRotateHandle.js:9]: We always use trailing commas. It is our styling preference and we try to keep it consistent.
```suggestion
      iconAnchor: [16, 16],
```

File: src/edit/handles/FreeRotateHandle.js [hunk 3]
```diff
@@ -1,44 +1,48 @@
...
+        "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAklEQVR4AewaftIAAAHiSURBVMXBa3HbShgA0PMp/1sCCo8oCEpgTaCXgIXAJiDzyCJoAUTm4UVQAns1Y8+snWnTvJyeE16hkjDgDrfoNTMKcpC9UPiLSo8JyetkjEHxjPCMyoS199kFoz8Iv1HpMaN3qWDCHoegOKkkRwnJpRmroHgiPFEZ8IBekzEGxQtUEhKSS/fB7Ew4U+lxcGkVZG9QWWPSFAxBcdK59KApuA+yNwp2uEdx1GN25sZJZULSfAtm77SlbNjju6MvG75u+WHRWVR6rDVjMPsgwYyVZl3pLTpHkyYHOx8syMiayaJzlDTZ9YyaZNFVkiYH2ZUEBcVJJXVImuz6Js3Qofe59pq7DoOTILu+g+a288mCouk7/1iH4qTS+2QdDppbV1ZJmrnDXnPnc5UOs2Z0fUmTuyBr+krvSioJyUmQO0dZM7mepMkWnaNRkyrJB6uskTSjxY3Fll8bvmJwlDb83FJ8gMqAB80uyBY3Trb82PAfvjj6vuHnluIdKgMeNXOwctK5NKBoHitrb1RJeHRp5Ux4ojLg0aWMHGQvUOkxIWkKVsHsTPiNSo8HDC5lZIsgO6n0uMUdRvQuFQxB8UR4RmXC2vvsgtEfhL+o9JiQvE7GGBTPCK9QSUjoMWgKDthjDrIX+h/k0I7gth6N5gAAAABJRU5ErkJggg==",
```
Comment (sashadev-sky) [inline src/edit/handles/FreeRotateHandle.js:7]: Again please put back the single quotes for this

```suggestion
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAklEQVR4AewaftIAAAHiSURBVMXBa3HbShgA0PMp/1sCCo8oCEpgTaCXgIXAJiDzyCJoAUTm4UVQAns1Y8+snWnTvJyeE16hkjDgDrfoNTMKcpC9UPiLSo8JyetkjEHxjPCMyoS199kFoz8Iv1HpMaN3qWDCHoegOKkkRwnJpRmroHgiPFEZ8IBekzEGxQtUEhKSS/fB7Ew4U+lxcGkVZG9QWWPSFAxBcdK59KApuA+yNwp2uEdx1GN25sZJZULSfAtm77SlbNjju6MvG75u+WHRWVR6rDVjMPsgwYyVZl3pLTpHkyYHOx8syMiayaJzlDTZ9YyaZNFVkiYH2ZUEBcVJJXVImuz6Js3Qofe59pq7DoOTILu+g+a288mCouk7/1iH4qTS+2QdDppbV1ZJmrnDXnPnc5UOs2Z0fUmTuyBr+krvSioJyUmQO0dZM7mepMkWnaNRkyrJB6uskTSjxY3Fll8bvmJwlDb83FJ8gMqAB80uyBY3Trb82PAfvjj6vuHnluIdKgMeNXOwctK5NKBoHitrb1RJeHRp5Ux4ojLg0aWMHGQvUOkxIWkKVsHsTPiNSo8HDC5lZIsgO6n0uMUdRvQuFQxB8UR4RmXC2vvsgtEfhL+o9JiQvE7GGBTPCK9QSUjoMWgKDthjDrIX+h/k0I7gth6N5gAAAABJRU5ErkJggg==',
```

File: src/edit/handles/FreeRotateHandle.js [hunk 4]
```diff
@@ -1,44 +1,48 @@
...
     } else {
       overlay.scaleBy(1);
     }
   },
 
-  updateHandle: function() {
+  updateHandle() {
     this.setLatLng(this._handled.getCorner(this._corner));
-  },
+  }
```
Comment (sashadev-sky) [inline src/edit/handles/FreeRotateHandle.js:43]: Please put back the trailing comma here.
```suggestion
  },
```

File: src/edit/handles/FreeRotateHandle.js [hunk 5]
```diff
@@ -1,44 +1,48 @@
...
 });
 
-L.freeRotateHandle = function(overlay, idx, options) {
+L.freeRotateHandle = (overlay, idx, options) => {
```
Comment (sashadev-sky) [inline src/edit/handles/FreeRotateHandle.js:46]: nice! :)

**Comment by sashadev-sky** [comment]
Also as far as your PR title, totally fine this time around but in the future I would encourage you to make a more specific title. Even using the original Issue Title would have worked.

File: src/edit/handles/FreeRotateHandle.js (hunk 1 above)
Comment (Manwej) [reply src/edit/handles/FreeRotateHandle.js:3]: got it, frankly that was visual studio code auto formatting. should have double checked.

File: src/edit/handles/FreeRotateHandle.js (hunk 2 above)
Comment (Manwej) [reply src/edit/handles/FreeRotateHandle.js:9]: will do, what are they for anyways?

**Comment by Manwej** [comment]
not 100% how to resolve those changes other than making a new pull request.

File: src/edit/handles/FreeRotateHandle.js (hunk 1 above)
Comment (sashadev-sky) [reply src/edit/handles/FreeRotateHandle.js]: yeah i mean, one way isn't better than the other, it's just preference. The essential thing is to maintain consistency.

File: src/edit/handles/FreeRotateHandle.js (hunk 2 above)
Comment (sashadev-sky) [reply src/edit/handles/FreeRotateHandle.js]: trailing commas also are just a preference. Again for maintaining clear standards / conventions for new contributors

**Comment by sashadev-sky** [comment]
Also 1 future tip for making PR's - you want to give the title something a little more specific, for ex., something similar to the original issue's title would have worked in this case

**Comment by Manwej** [comment]
@sashadev-sky Thanks a lot for the guidance and tips. My Twitter is @Manwej. All the best

**Pull request merged**